internal/api/handler/handler.go

```diff
@@ -337,6 +338,64 @@ func (h *Handler) GetWorkouts(c *gin.Context) {
 	c.JSON(http.StatusOK, workoutList)
 }
 
+func (h *Handler) GetWorkoutsForUser(c *gin.Context) {
```

Comment:
This one might be redundant, will have to look into if I can refactor the previous GetWorkouts function to dynamically switch which SQL query it calls